### ISSUE TITLE
Color picker: focus loss when pressing Escape after values are changed

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#50876]
### Description

Initial consideration:
When using the keyboard to pick a color in the color picker, the only way to _set_ the new color and close the color picker is by pressing the Escape key. I don't think this is right, as the Escape key is universally used to _cancel_ the current operation. Pressing Escape should close the color picker and _not set_ any color.
Instead, there should be another way to _set_ the color: preferably a dedicated 'Apply' button. I'll create a separate issue.

That said, when the color picker gets closed, whether a color has been set or not, focus should move back to the color button that was clicked to open the color picker. This appears to fail but only after the color value has been changed.

Animated GIF to better illustrate:

![color picker focus loss](https://github.com/WordPress/gutenberg/assets/1682452/2cac2d52-b25c-44de-9449-322b630d095e)

The focus loss occurs also when changing the value in the HEX input field:

![color picker focus loss 2](https://github.com/WordPress/gutenberg/assets/1682452/710ffd4d-a042-4bbe-8f67-e1a56ce5214b)

### Step-by-step reproduction instructions

- Go to the Site Editor.
- Edit the color palette.
- Click a color button to open the color picker.
- Press the Escape key to close the color picker.
- Focus is moved back to the color button that was clicked previously. So far so good.
- Open again the color picker.
- Change the color value:
  - Either by focusing one of the sliders and by using the arrow keys to change their value.
  - Or by manually entering a new value in the Hex color input field.
- Press the Escape key to set the new color and close the picker.
- Observe the color button that was clicked to open the picker doesn't receive focus.
- Press the Tab key.
- Observe the tab sequence starts from scratch from the document root.

Note: A similar focus loss occurs also with the color picker in HSL and RGB modes.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
